src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -1463,13 +1465,26 @@ void scrollElementToFirstCellAndSendKeys(WebElement spreadsheetElement, String v
         private static final String START_OCCURRED_ATTRIBUTE = "__ajaxStartOccurred__";
 
         /**
-         * Registers `ajaxStart` and `ajaxStop` handlers to track the state of an AJAX request.
+         * Waits until there is no ongoing jQuery AJAX requests.
+         */
+        void waitForNoActiveAjaxRequests() {
+            // `$.active` is a counter for holding the number of active AJAX requests
+            // but is not documented because it is used by JQuery internally.
+            // see https://stackoverflow.com/questions/3148225/jquery-active-function/3148506#3148506
+            waitFor(driver -> (Boolean) ((JavascriptExecutor) driver).executeScript("return $.active === 0"));
+        }
+
+        /**
+         * Waits for all AJAX requests to complete if any is present and registers `ajaxStart` and `ajaxStop` handlers to
+         * track the state of an AJAX request.
          *
          * @throws IllegalStateException if the handlers are already registered in the document
          */
-        void registerHandlers() {
+        void waitForAjaxIfPresentAndRegisterHandlers() {
```

Comment:
I realized maybe `waitForAjaxIfPresent**Then**RegisterHandlers` makes it clearer it's a sequence. As with others like `waitForConfirmationModal**Then**ClickOk`. Just mentioning this though, up to you to decide.